src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -680,24 +798,120 @@ def get_observation(self, event: Event) -> pd.DataFrame:
+            "last_name_id",
+            "age",
+            "date_of_birth",
+            "sex",
+            "ssn",
+            "address_id",
+            "housing_type",
+        ]:
+            df[col] = df["dependent_id"].map(pop_full[col])
 
-        for col, s in [
-            ["eligible_dependent_id_list", s_eligible_dependent_id_list],
-            ["eligible_dependent_address_id_list", s_eligible_dependent_address_id_list],
+        # delete extraneous columns
+        for col in [
+            "in_united_states",
+            "alive",
+            "guardian_1",
+            "guardian_2",
+            "last_year_income",
+            "race_ethnicity",
+            "housing_type",
+            "tracked",
         ]:
-            df_w2[col] = df_w2["simulant_id"].map(s)
+            del df[col]
 
-        # re-initialize income-to-date series for next year of income counting
-        self.income_to_date = empty_income_series()
+        return df
 
-        return df_w2[self.output_columns]
+
+class Tax1040Observer(BaseObserver):
+    """Class for observing columns relevant to 1040 tax data (that is not
+    already recorded by W2 or dependents observer)
+    """
+
+    INPUT_VALUES = ["income", "household_details", "business_details"]
+    ADDITIONAL_INPUT_COLUMNS = [
+        "alive",
+        "tracked",
+        "ssn",
+        "relation_to_household_head",
+    ]
+    OUTPUT_COLUMNS = [
+        "first_name_id",
+        "middle_name_id",
+        "last_name_id",
+        "age",
+        "date_of_birth",
+        "sex",
+        "ssn",
+        "address_id",
+        "relation_to_household_head",
+        "housing_type",
+        "tax_year",
+    ]
+
+    def __init__(self, w2_observer):
+        super().__init__()
+        self.w2_observer = w2_observer
+
+    def __repr__(self):
+        return f"Tax1040Observer()"
+
+    @property
+    def name(self):
+        return f"tax_1040_observer"
+
+    @property
+    def input_columns(self):
+        return self.DEFAULT_INPUT_COLUMNS + self.ADDITIONAL_INPUT_COLUMNS
+
+    @property
+    def input_values(self):
+        return self.INPUT_VALUES
+
+    @property
+    def output_columns(self):
+        return self.OUTPUT_COLUMNS
+
+    def setup(self, builder: Builder):
+        super().setup(builder)
+        self.clock = builder.time.clock()
+
+    def to_observe(self, event: Event) -> bool:
+        """Observe if April 15 falls during this time step"""
+        tax_date = dt.datetime(event.time.year, 4, 15)
+        return self.clock() < tax_date <= event.time
+
+    def get_observation(self, event: Event) -> pd.DataFrame:
+        pop = self.population_view.get(event.index)
+        household_details = self.pipelines["household_details"](pop.index)
+        pop[["address_id", "housing_type"]] = household_details[
+            ["address_id", "housing_type"]
+        ]
+
+        # add derived columns
+        pop["tax_year"] = event.time.year - 1
+
+        # delete extraneous columns
```

Comment:
Similar to above, why not `pop = pop[OUTPUT_COLUMNS]`?

Reply:
> Similar to above, why not pop = pop[OUTPUT_COLUMNS]?

Just bad taste on my part, I will change